phpunit/functional/NotificationTargetTest.php

```diff
@@ -638,4 +640,51 @@ public function testGetTargetsWithExclusions()
         $target = reset($targets);
         $this->assertEquals(getItemByTypeName('User', TU_USER, true), $target['users_id']);
     }
+
+    public function testAdminTargetInstall()
+    {
+        $this->login();
+
+        $notification = new \Notification();
+        $notifications = $notification->find();
+
+        $tables = require(GLPI_ROOT . '/install/empty_data.php');
+        foreach ($notifications as $notification) {
+            $notification_target = NotificationTarget::getInstanceByType($notification['itemtype'], $notification['event']);
+            $notification_target->addNotificationTargets(0);
+            $filtered = array_filter(
+                $tables["glpi_notificationtargets"],
+                function ($empty_data_array) use ($notification) {
+                    return isset($empty_data_array['items_id'], $empty_data_array['notifications_id']) &&
+                        $empty_data_array['items_id'] === '1' &&
+                        $empty_data_array['notifications_id'] === strval($notification['id']);
+                }
+            );
+            if (isset($notification_target->notification_targets["1_1"]) && $notification_target->notification_targets["1_1"] === "1_" . __("Administrator")) {
+                $this->assertNotEmpty($filtered);
+            } else {
+                $this->assertEmpty($filtered);
+            }
```

Comment:
You should hardcode notifications that are legitimate to not have the administrator as a target, and add a comment for each of them.
e.g.
```php
$has_admin_target = true;
if ($notification['itemtype'] === 'User' && $notification['event'] === 'passwordforget') {
    // Only the user is able to receive its password recovery token
    $has_admin_target = false
} elseif ...

$this->assertSame($has_admin_target, array_key_exists('1_1', $notification_target->notification_targets));
```